src/libecalc/core/models/compressor/train/base.py

```diff
@@ -101,17 +101,20 @@ def evaluate_rate_ps_pd(
         )
 
         if self.data_transfer_object.calculate_max_rate:
+            # only calculate max standard rate for time steps with valid input
+            valid_idx = np.where([failure_status is None for failure_status in input_failure_status], 1, 0)
             if isinstance(self.data_transfer_object, dto.VariableSpeedCompressorTrainMultipleStreamsAndPressures):
-                max_standard_rate = self.get_max_standard_rate_per_stream(
-                    suction_pressures=suction_pressure,
-                    discharge_pressures=discharge_pressure,
-                    rates_per_stream=rate,
+                max_standard_rate_for_valid_idx = self.get_max_standard_rate_per_stream(
+                    suction_pressures=suction_pressure[valid_idx],
+                    discharge_pressures=discharge_pressure[valid_idx],
+                    rates_per_stream=rate[:, valid_idx],
                 )
             else:
-                max_standard_rate = self.get_max_standard_rate(
-                    suction_pressures=suction_pressure,
-                    discharge_pressures=discharge_pressure,
+                max_standard_rate_for_valid_idx = self.get_max_standard_rate(
+                    suction_pressures=suction_pressure[valid_idx],
+                    discharge_pressures=discharge_pressure[valid_idx],
                 )
+            max_standard_rate = np.where(valid_idx, max_standard_rate_for_valid_idx, np.nan)
```

Comment:
```suggestion
            INVALID_INPUT = np.nan
            INVALID_MAX_RATE = INVALID_INPUT
            max_standard_rate = np.where(valid_idx, max_standard_rate_for_valid_idx, INVALID_MAX_RATE)
```

open for other suggestions

Reply:
INVALID_INPUT should then be in a errors.py file or similar, where we have code related to invalid data etc